### ISSUE TITLE
[Feature Request] Extend the map to switch locations from it by making it interactive

### ISSUE THREAD

**Issue by Adoliin** [mullvad/mullvadvpn-app#2901]
### Interactive Map
The preexisting map is okay but a nice feature would be:
- Switch locations by pressing on cities and countries on the map instead of pressing `Switch location` and keep looking for a country / city.

- Also this gives a visual cue to approximate latency based on how much far the selected country is from the user actual country.

**Comment by faern** [comment]
Physical distance is not a very good measurement of latency. In the best of worlds it would be of course. The traffic would take a straight path to the destination at the speed of light. But in reality there can be bad peering between providers and throttled connections between some data centers etc. This is the main reason why we don't provide any "nearest server" or "fastest server" option. Because there is no good reliable way of figuring it out programmatically without being too noisy on the network.

I agree it could be cool to show the available locations on the map. But the map is only intended as background visualization, not to be interactive really. It would mess a lot with the actual clickable stuff drawn above it. But I'll mark this as a feature request for now.

**Comment by MagicLike** [comment]
I also imagine some kind of a interactive map or at least a animated map showing the location of the server(s) currently connected to (or even draw sine kind of connections to see where the traffic is going) as a great use of the already existing map and just a cool visualisation feature.
Are there any updates on this or plans to implement something like that?

**Comment by faern** [comment]
Since the earth is a sphere, all two pairs of points can't be shown at once. If you are in Australia and connect to Sweden, we can't possibly show both the origin location and the server location in one view.

We do show the location of the currently connected server. Or if you are disconnected, your apparent location given a GeoIP lookup against our own IP location service.

**Comment by MagicLike** [comment]
> Since the earth is a sphere, all two pairs of points can't be shown at once. If you are in Australia and connect to Sweden, we can't possibly show both the origin location and the server location in one view.

I mean if you show the earth as a sphere instead of showing it as a map, the spere could theoretically slowly rotate.
But if it was shown solely as a map, it would be easy to visualise

> We do show the location of the currently connected server. Or if you are disconnected, your apparent location given a GeoIP lookup against our own IP location service.

I mean, wouldn't it be possible to literally connect the dots here? :P

**Comment by faern** [comment]
Yes, if it wasn't a sphere. And it is. We recently changed the map implementation from a flat map to a 3d globe. You are correct that it would be possible to "slowly rotate" but I don't see how that would look good overall. Would we pan back and fourth between your location and the exit?

**Comment by MagicLike** [comment]
> Yes, if it wasn't a sphere. And it is. We recently changed the map implementation from a flat map to a 3d globe.

Oh I did not notice that 🥴

> You are correct that it would be possible to "slowly rotate" but I don't see how that would look good overall. Would we pan back and fourth between your location and the exit?

Either that or just let it slowly rotate 360° (or try to get both points into the frame and hild it there/zoom out, which on a second thought seems like a bad idea)
